src/main/java/org/mule/extension/db/internal/resolver/query/AbstractQueryResolver.java

```diff
@@ -108,6 +108,10 @@ private QueryTemplate resolveQueryTemplate(QueryTemplate queryTemplate, Map<Inte
       DbType type = paramTypes.get(originalParam.getIndex());
       QueryParam newParam;
 
+      if (type == null) {
+        throw new IllegalArgumentException("Unknown parameter type of " + originalParam.getName());
```

Comment:
Throwing a plain IllegalArgumentException doesn't comply with the Extension Error mechanism, this will produce in runtime to throw a `MULE:UNKNOWN`. 
All DML Operations throws [these errors](https://github.com/mulesoft/mule-db-connector/blob/master/src/main/java/org/mule/extension/db/internal/operation/OperationErrorTypeProvider.java), the exception should communicate one of those, in this case, a `QUERY_EXECUTION` error, this can be achieved throwing a [`QueryExecutionException.java`](https://github.com/mulesoft/mule-db-connector/blob/master/src/main/java/org/mule/extension/db/api/exception/connection/QueryExecutionException.java)